test/e2e/framework/pod/resource.go

```diff
@@ -452,7 +452,7 @@ func CreateExecPodOrFail(client clientset.Interface, ns, generateName string, tw
 		}
 		return retrievedPod.Status.Phase == v1.PodRunning, nil
 	})
-	expectNoError(err)
+	expectNoError(err, "failed to create new exec pod in namespace: %s", ns)
```

Comment:
having the pod name here and on line 444 would have been one idea more useful too.

Reply:
agree, but it is an `exec pod` , I mean, the name is randomly generated and typically you only use one per test :shrug:

Reply:
the namespace is also random.

Reply:
but yes, it does not matter much as long as the error message can be traced back to the line of code.

Reply:
the main goal is to have a different message so this query does not aggregate different errors into the same bucket
https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=Services%20should%20be%20rejected%20when%20no%20endpoints%20exist